packages/api/src/controllers/asset.ts

```diff
@@ -1089,11 +1110,64 @@ app.delete("/:id", authorizer({}), async (req, res) => {
   if (!req.user.admin && req.user.id !== asset.userId) {
     throw new ForbiddenError(`users may only delete their own assets`);
   }
+
+  if (asset.status.phase === "deleting" || asset.deleted) {
+    throw new BadRequestError(`Asset is already deleted`);
+  }
+
   await req.taskScheduler.deleteAsset(asset);
   res.status(204);
   res.end();
 });
 
+app.put("/:id/restore", authorizer({}), async (req, res) => {
```

Comment:
I believe this should be a `POST` which AFAIK is the general fallback in REST for an "action" on a resource, which is what being triggered here.